python2/koans/about_iteration.py

```diff
@@ -75,28 +75,29 @@ def multiply(self, accum, item):
 
     def test_reduce_will_blow_your_mind(self):
         result = reduce(self.add, [2, 3, 4])
-        self.assertEqual(__, result)
+        self.assertEqual(9, result)
 
         result2 = reduce(self.multiply, [2, 3, 4], 1)
-        self.assertEqual(__, result2)
+        self.assertEqual(24, result2)
 
         # Extra Credit:
         # Describe in your own words what reduce does.
-
+        
+        # Apply the function to every member of the list, the final return is what the final iteration of the function outputs
```

Comment:
Yeah, but it's a bit more specific than that, right? how does it differ from `map`?

Reply:
The way in which `reduce` works and differs from is that it receives a function and a sequence(just as map) but it returns just a single value.

The way that this value is calculated is by taking the first two members of the `sequence` and then executing the function, after that, the result of that first execution is used with the next value of the sequence, this process gets repeated until the last member of the sequence is taken to account.